src/FSharp Examples/SequenceToSequence.fs

```diff
@@ -0,0 +1,273 @@
+module TorchSharp.Examples.SequenceToSequence
+
+open System
+open System.IO
+open System.Linq
+open System.Diagnostics
+open System.Collections.Generic
+
+open TorchSharp
+open TorchSharp.Tensor
+open TorchSharp.NN
+
+open type TorchSharp.NN.Modules
+
+open TorchSharp.Examples
+
+/// <summary>
+/// This example is based on the PyTorch tutorial at:
+/// 
+/// https://pytorch.org/tutorials/beginner/transformer_tutorial.html
+///
+/// It relies on the WikiText2 dataset, which can be downloaded at:
+///
+/// https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-2-v1.zip
+///
+/// </summary>
+
+let emsize = 200L
+let nhidden = 200L
+let nlayers = 2L
+let nheads = 2L
+let dropout = 0.2
+let bptt = 32L
+
+let batch_size = 64L
+let eval_batch_size = 256L
+
+let epochs = 50
+
+let logInterval = 200
+
+let cmdArgs = Environment.GetCommandLineArgs()
+
+let datasetPath = Path.Join(Environment.GetFolderPath(Environment.SpecialFolder.DesktopDirectory), "..", "Downloads", "wikitext-2-v1")
+
+Torch.SetSeed(1L)
+
+let hasCUDA = Torch.IsCudaAvailable()
+
+let device = if hasCUDA then Device.CUDA else Device.CPU
+
+let criterion x y = Functions.cross_entropy_loss(reduction=Reduction.Mean).Invoke(x,y)
+
+type PositionalEncoding(dmodel, maxLen) as this =
+    inherit CustomModule("PositionalEncoding")
+
+    let dropout = Dropout(dropout)
+    let mutable pe = Float32Tensor.zeros([| maxLen; dmodel|])
+
+    do
+        let position = Float32Tensor.arange(0L.ToScalar(), maxLen.ToScalar(), 1L.ToScalar()).unsqueeze(1L)
+        let divTerm = (Float32Tensor.arange(0L.ToScalar(), dmodel.ToScalar(), 2L.ToScalar()) * (-Math.Log(10000.0) / (float dmodel)).ToScalar()).exp()
+
+        let NULL = System.Nullable<int64>()
+        pe.[TorchTensorIndex.Ellipsis, TorchTensorIndex.Slice(0L, NULL, 2L)] <- (position * divTerm).sin()
+        pe.[TorchTensorIndex.Ellipsis, TorchTensorIndex.Slice(1L, NULL, 2L)] <- (position * divTerm).cos()
+
+        pe <- pe.unsqueeze(0L).transpose(0L,1L)
+
+        this.RegisterComponents()
+
+    override _.forward(t) =
+        let NULL = System.Nullable<int64>()
```

Comment:
You can actually just use `System.Nullable()` here, as type inference will figure out the specific type. `null` cannot be passed in directly, however.